src/expr.rs

```diff
@@ -816,10 +814,8 @@ fn rewrite_if_else(context: &RewriteContext,
         _ => " ",
     };
     let mut result = format!("if{}{}{}{}",
-                             between_if_cond_comment.as_ref().map_or(" ", |str| &**str),
-                             pat_expr_string,
-                             after_cond_comment.as_ref().map_or(after_sep, |str| &**str),
-                             if_block_string);
+                             between_if_cond_comment.as_ref().map_or(" ", |str| &**str), pat_expr_string,
+                             after_cond_comment.as_ref().map_or(after_sep, |str| &**str), if_block_string);
```

Comment:
Yes, that seems correct to me. I.e., it effectively ignores the first argument, and lays out the rest as if it didn't exist, other than causing a newline. In particular, it should follow any user preference for how the args are laid out, other than as they apply to the first one.